srcCxx/shape_main.cxx

```diff
@@ -959,15 +964,19 @@ class ShapeApplication {
             sprintf(parameter, "'%s'",  options->color);
             StringSeq_push(cf_params, parameter);
             cft = dp->create_contentfilteredtopic(options->topic_name, topic, "color MATCH %0", cf_params);
-            logger.log_message("    ContentFilterTopic = color MATCH " + std::string(parameter), Verbosity::DEBUG);
+            logger.log_message("    ContentFilterTopic = \"color MATCH "
+                + std::string(parameter) + std::string("\""), Verbosity::DEBUG);
 #elif defined(TWINOAKS_COREDX) || defined(OPENDDS)
             StringSeq_push(cf_params, options->color);
             cft = dp->create_contentfilteredtopic(options->topic_name, topic, "color = %0", cf_params);
-            logger.log_message("    ContentFilterTopic = color = " + std::string(options->color), Verbosity::DEBUG);
+            logger.log_message("    ContentFilterTopic = \"color = "
+                    + std::string(options->color) + std::string("\""),
+                Verbosity::DEBUG);
 #elif defined(EPROSIMA_FAST_DDS)
             cf_params.push_back(std::string("'") + options->color + std::string("'"));
             cft = dp->create_contentfilteredtopic(std::string(options->topic_name) + "_filtered", topic, "color = %0", cf_params);
-            logger.log_message("    ContentFilterTopic = color = " + std::string(options->color), Verbosity::DEBUG);
+            logger.log_message("    ContentFilterTopic = \"color = "
+                + std::string(options->color) + std::string("\""), Verbosity::DEBUG);
```

Comment:
Just realized that, since we are adding single quotes to the input parameter, it would be better to use the parameter passed to `create_contentfilteredtopic`:

```suggestion
                + cf_params[0] + std::string("\""), Verbosity::DEBUG);
```

Reply:
done